### PR TITLE
feat: support tickets history

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes gravitee-io/issues#4435

### CODE REVIEW

File: src/main/java/io/gravitee/repository/management/model/Ticket.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("User{");
+        sb.append("id='").append(id).append('\'');
+        sb.append(", senderId='").append(fromUser).append('\'');
```
Comment (aelamrani) [inline src/main/java/io/gravitee/repository/management/model/Ticket.java:150]: senderId => fromuser

Comment (ytvnr) [reply src/main/java/io/gravitee/repository/management/model/Ticket.java:150]: resolved !

File: src/main/java/io/gravitee/repository/management/model/Ticket.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    private String api;
+
+    /**
+     * The application id concerned by the ticket
+     */
+    private String application;
+
+    /**
+     * The user first name
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/management/model/Ticket.java:48]: not the right comment ;)

File: src/main/java/io/gravitee/repository/management/model/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+     */
+    private String subject;
+
+    /**
+     * The user last name
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/management/model/Ticket.java:53]: not the right comment ;)

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: src/main/java/io/gravitee/repository/management/api/search/TicketCriteria.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class TicketCriteria {
+
+    private String fromUser;
+
+    private String api;
+
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/management/api/search/TicketCriteria.java:29]: why not allow search by application?

File: src/main/java/io/gravitee/repository/management/model/Ticket.java (hunk 2 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/repository/management/model/Ticket.java:48]: done

File: src/main/java/io/gravitee/repository/management/model/Ticket.java (hunk 3 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/repository/management/model/Ticket.java:53]: done

File: src/main/java/io/gravitee/repository/management/api/search/TicketCriteria.java (hunk 4 above)
Comment (ytvnr) [reply src/main/java/io/gravitee/repository/management/api/search/TicketCriteria.java:29]: done

**Review by phiz71 — APPROVED** [review]

**Pull request merged**